### PR TITLE
[Feature:TAGrading] Rubric component wraparound

### PR DESCRIPTION
**Pull request by DefinitiveAbove**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
Clicking the up arrow at the top of the grading rubric does not wrap around to the bottom, clicking the bottom arrow at the bottom of the grading rubric does not wrap around to the top.

### What is the new behavior
Works on #4739 
Clicking the up arrow at the top of the grading rubric wraps around to the bottom, clicking the bottom arrow at the bottom of the grading rubric wraps around to the top.

### CODE REVIEW

File: site/public/js/ta-grading.js [hunk 1]
```diff
@@ -967,9 +967,11 @@ registerKeyHandler({name: "Open Next Component", code: 'ArrowDown'}, function(e)
   }
   else if (openComponentId === getComponentIdByOrder(numComponents - 1)) {
     // Last component is open, scroll to general comment for easier access
```
Comment (MasterOdin) [inline site/public/js/ta-grading.js:969]:
```suggestion
    // Last component is open, close it and then open and scroll to first component
```

File: site/public/js/ta-grading.js [hunk 2]
```diff
@@ -995,8 +997,13 @@ registerKeyHandler({name: "Open Previous Component", code: 'ArrowUp'}, function(
       scrollToOverallComment();
   }
   else if (openComponentId === getComponentIdByOrder(0)) {
-    // First component is open, so close it
-    closeAllComponents(true);
+    // First component is open, so open the last one
```
Comment (MasterOdin) [inline site/public/js/ta-grading.js:1000]:
```suggestion
    // First component is open, close it and then open and scroll to the last one
```

File: site/public/js/ta-grading.js (hunk 1 above)
Comment (MasterOdin) [reply site/public/js/ta-grading.js:969]: Need to make sure we keep comments up-to-date with what the code behavior is :)

**Review by shailpatels — CHANGES_REQUESTED** [review]
Works fine for normal rubrics for individual and team gradeables.

For peer gradeables there is an issue where pressing the up arrow at the first component or 
pressing the down arrow at the last component (i.e trying to loop around) causes the alert message:

`Are you sure you want to delete the custom mark?` consistently 

The following JS console error also is logged:

`Uncaught (in promise) undefined`

**Review by KKhaghani — APPROVED** [review]
Tested the new behavior after the peer component fix, looks good.

**Review by shailpatels — APPROVED** [review]
Changes look good and works as described

**Pull request merged**